src/node/node_state.h

```diff
@@ -541,7 +542,9 @@ namespace ccf
           }
 
           // Set network secrets, node id and become part of network.
-          if (resp.node_status == NodeStatus::TRUSTED)
+          if (
```

Comment:
As discussed, we need to make sure that we cover all places where we currently nodes as `TRUSTED`, e.g. in `secret_broadcast.h`.

Reply:
I took another look and added learners in `secret_broadcast.h` and in another place in `node_frontend.h`. I think this covers all the cases we need, but I'd be grateful if you could take another look to make sure I'm not giving secrets to nodes who aren't supposed to get them.